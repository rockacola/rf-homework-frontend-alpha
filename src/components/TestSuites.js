import React, { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import { getTestSuites } from "../endpoints/api";
import { TestSuiteRow } from "./TestSuiteRow";

export const TestSuites = () => {
  // Fetch data
  // Detect if data is ready
  // Render line by line

  const [testSuites, setTestSuites] = useState([]);

  useEffect(() => {
    console.log("TestSuites onmount");

    const act = async () => {
      const res = await getTestSuites();
      setTestSuites(res);
    };
    act();
  }, []);

  // TODO: have a hydrating state
  return (
    <div>
      <div>TestSuites. count: {testSuites.length}</div>
      <div style={{ paddingTop: "10px", paddingBottom: "20px" }}>
        <Link to="/test_suites/create">[Create New]</Link>
      </div>
      <div>
        <table border="1" cellspacing="0" cellpadding="0">
          <thead>
            <th style={{ width: "50px" }}></th>
            <th style={{ width: "250px" }}>Name</th>
            <th style={{ width: "450px" }}>Count</th>
            <th></th>
          </thead>
          <tbody>
            {testSuites.map((ts) => (
              <TestSuiteRow testSuite={ts} />
            ))}
          </tbody>
        </table>
      </div>
    </div>
  );
};
