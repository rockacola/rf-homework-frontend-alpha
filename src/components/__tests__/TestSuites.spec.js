import React from "react";
import renderer from "react-test-renderer";
import { TestSuites } from "../TestSuites";

describe("TestSuites", () => {
  let comp, props;

  describe("when empty", () => {
    beforeEach(() => {
      props = { testSuites: [] };
      comp = renderer.create(<TestSuites {...props} />);
    });

    it("renders", () => {
      expect(comp).toMatchSnapshot();
    });
  });
});
