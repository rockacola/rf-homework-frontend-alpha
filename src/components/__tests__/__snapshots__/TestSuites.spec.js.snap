// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestSuites when empty renders 1`] = `
<table
  border="1"
  cellPadding="0"
  cellSpacing="0"
>
  <thead>
    <tr>
      <th
        style={
          Object {
            "width": "50px",
          }
        }
      />
      <th
        style={
          Object {
            "width": "250px",
          }
        }
      >
        Name
      </th>
      <th
        style={
          Object {
            "width": "450px",
          }
        }
      >
        Count
      </th>
      <th />
    </tr>
  </thead>
  <tbody />
</table>
`;
